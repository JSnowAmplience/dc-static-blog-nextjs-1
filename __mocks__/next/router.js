module.exports = {
  useRouter: function() {
    return { asPath: 'mocked-path' };
  }
};
