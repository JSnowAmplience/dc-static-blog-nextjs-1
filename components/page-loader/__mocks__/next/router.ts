export const mockRouterOn = jest.fn();
export const Router = { events: { on: mockRouterOn } };
export const useRouter = (): { asPath: string; query: {} } => ({ asPath: 'mocked-path', query: {} });
