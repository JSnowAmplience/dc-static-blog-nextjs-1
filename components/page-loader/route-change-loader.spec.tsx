/* eslint-env jest */
import renderer from 'react-test-renderer';
import RouteChangeLoader from './route-change-loader';
// @ts-ignore
import { mockRouterOn } from 'next/router';
import { shallow } from 'enzyme';

describe('RouteChangeLoader', () => {
  test('renders route change loader when loading', async () => {
    const wrapper = shallow(<RouteChangeLoader />);
    wrapper.setState({ isLoading: true });
    expect(wrapper.html()).toMatchSnapshot();
  });
  test('renders route change loader when not loading', async () => {
    const wrapper = shallow(<RouteChangeLoader />);
    wrapper.setState({ isLoading: false });
    expect(wrapper.html()).toMatchSnapshot();
  });
  test('renders route change loader with correct event listeners', async () => {
    await renderer.create(<RouteChangeLoader />);
    expect(mockRouterOn.mock.calls[0][0]).toEqual('routeChangeStart');
    expect(mockRouterOn.mock.calls[1][0]).toEqual('routeChangeComplete');
    expect(mockRouterOn.mock.calls[2][0]).toEqual('routeChangeError');
  });
});
