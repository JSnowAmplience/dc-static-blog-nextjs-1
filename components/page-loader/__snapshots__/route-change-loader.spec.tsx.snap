// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RouteChangeLoader renders route change loader when loading 1`] = `"<div class=\\"jsx-538995977 page-loading\\"><div class=\\"jsx-538995977 spinner\\"></div></div>"`;

exports[`RouteChangeLoader renders route change loader when not loading 1`] = `""`;
